### ISSUE TITLE
SAB Reporting definitions and descriptions updates

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#484]
In GitLab by @kdjstudios on Sep 20, 2017, 12:26

Hello Team,

This thread will be used to help us update the SAB Reporting definitions. Gary and I had a meeting to start updating the reporting definitions, however it lead to more unanswered questions about the reports then updating the descriptions. 

I have attached the notes Gary and I started to add to the first few reports. However, most of these questions need to be answered (For all reports, not just the ones we have notes on) by someone with the technical code knowledge of how the report functions. Thus, why we are needing your input for all reports, not just the ones we highlighted, to create Extremely detailed descriptions about the report and its contents. This would include:
* Descriptions of all variations that report can be ran as
* Descriptions of what each filter option will alter or perform on the report
* Description of the reports purpose and general use
* Examples of when and how to use the report effectively, also how not to use the report.  *This we may need to enlist operations for help*
* More to come with further discussion 

We also found quite a few common improvements and requests that most have previously been mentioned before either in previous Gitlab tickets or we will create new ones separately as approved. You may ignore for now those notes in the attachment.

[Report_Definitions_-_Notes_9-20-17.docx](/uploads/0f85c540a64cb582b782e343f8eb8450/Report_Definitions_-_Notes_9-20-17.docx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 20, 2017, 12:28

My initial question, is would we like to use this thread as overview checklist for all accounts and to create a new gitlab ticket for each report to address its details separately?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 21, 2017, 08:42

@smasih @pchaudhary @tim.traylor @abakshi 

This needs to be completed by next Tuesday 9/26/17 end of day; before the Gary's next available meeting.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 22, 2017, 08:31

@smasih @tim.traylor @pchaudhary @abakshi 

Gary and I had to reschedule the meeting for Monday 9/25/17.  I was hoping to have some updates on this topic already today, as both Tim and I stressed the importance of this per Gary.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 24, 2017, 08:22

@kdjstudios  @abakshi 

update on reports 

[Book1__1_.xlsx](/uploads/11b862b83c501499212dc5de285dbcad/Book1__1_.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 25, 2017, 07:09

@kdjstudios @tim.traylor 

Updated Sheet on All SAB Reports

[Book1.xlsx](/uploads/510931ac3a63fbb4b383816e3842aba9/Book1.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2017, 08:32

@smasih 

Thank you for the file, I am starting to review it now and will also review this with Gary during our meeting.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2017, 12:54

@smasih 

Thank you for the definition it was almost exactly what we are looking for. I will be using the information to put together a new report definition section. I may be asking for further details on reports that still only have a basic description or purpose. 

Just a heads up for everyone, I will be opening a ticket for each report that took over 10 seconds to load. and I also will open tickets for each report that failed to run.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 3, 2017, 14:46

@smasih @tim.traylor 

I have created individual tickets for all reports that I saw had and issue or took over 10 seconds to load. I would estimate this to be about 80% of the reports.

The last uploaded spreadsheet is still missing lots of descriptions. Mainly on what each filter option will perform or alter on each report. We have for most reports a very basic description that reinforces that title of the report. In addition to improving those lackluster descriptions we will also need these improvements, as outlined in the ticket:
* Description of the reports purpose and general use
* Examples of when and how to use the report effectively, also how not to use the report.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 4, 2017, 02:00

@kdjstudios 

Sorry, but that is the description all we got.   
We can tell which data model is used and what's the logic behind the scenes but explaining its purpose in accounting won't be possible for us.

Reports we have at the moment is not very flexible, means they are very opinionated so there is only one way which is the right way.(No wrong way of using reports)

Thanks for other tickets and we will be adding a **Reports Milestone** to all of them just to sort out the reporting tickets.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Oct 4, 2017, 04:57

@smasih @kdjstudios Hi Sumeet, I agree that the usage of the report is not something you guys would address.  That is for our Ops and Accounting teams to do.

We do need for you to describe all of the filters though.  That was part of the original request Kyle made and is something you can describe based on the code.

Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 4, 2017, 08:28

@tim.traylor 

Should I be reaching out to Cori or whom for descriptions on the 'usage' of the report in real world?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 5, 2017, 07:36

@kdjstudios 

sure thing we will add the explanations on the filters.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 17, 2017, 10:19

@smasih @pchaudhary @tim.traylor 

The last uploaded file with descriptions, still seems to be incomplete there are about 30%-40% of the account filter options that do not have explanations of their purpose and function.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 30, 2017, 04:57

moving to next milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 31, 2017, 09:43

Moving priority and uncertain on the 11-11 release. Will discuss further as rebillables is completed.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 31, 2017, 09:43

Milestone removed

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 1, 2017, 13:15

I was able to start formatting the spreadsheet into a document guide. I plan on updating the report columns to also include brief definitions of each title. We are still needing further descriptions on what each filter option does; as right now we only have the title/label of the option it self. 

During this process I noticed more that all of the report descriptions are basically copy pasted from my report definitions, that Gary stated was not enough and was the purpose of this to further define them. @tim.traylor maybe this is something that the Operations/Accounting teams should be inputting on rather then the Aavaz team? 
> Per Summet's previous response: 

> Sorry, but that is the description all we got.

> We can tell which data model is used and what's the logic behind the scenes but explaining its purpose in accounting won't be possible for us.

In summary so far, we do have each report detailed out currently with the filter options and the expected data results; but we are still lacking in actual definition; example use; and purpose. 



* [Report_Definitions_from_Aavaz.xlsx](/uploads/776db0182fe54149548e9d0b3502c169/Report_Definitions_from_Aavaz.xlsx)

*  [SAB_Report_Definitions_Human_Readable_10-31-17.docx](/uploads/eb4bf14d86990dcaf00499cc6d1ef5ea/SAB_Report_Definitions_Human_Readable_10-31-17.docx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 12, 2019, 13:17

We should be good to close this ticket as the updated documentation is being complied in #1249

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 12, 2019, 13:17

closed
